ssl/ssl_local.h

```diff
@@ -294,8 +294,14 @@
  * flags because it may not be set to correct version yet.
  */
 # define SSL_CLIENT_USE_TLS1_2_CIPHERS(s)        \
-    ((!SSL_CONNECTION_IS_DTLS(s) && s->client_version >= TLS1_2_VERSION) || \
-     (SSL_CONNECTION_IS_DTLS(s) && DTLS_VERSION_GE(s->client_version, DTLS1_2_VERSION)))
+    ((!SSL_CONNECTION_IS_DTLS(s) \
+      && s->client_version >= TLS1_2_VERSION \
+      && (s->ssl.method->version == TLS_ANY_VERSION || \
+          s->version >= TLS1_2_VERSION)) \
+     || (SSL_CONNECTION_IS_DTLS(s) \
+         && DTLS_VERSION_GE(s->client_version, DTLS1_2_VERSION) \
+         && (s->ssl.method->version == DTLS_ANY_VERSION \
+             || DTLS_VERSION_GE(s->version, DTLS1_2_VERSION))))
```

Comment:
I'm not really sure what this is fixing. I suppose it is to make reasoning about `SSL_CLIENT_USE_SIGALG()` easier since it takes account of whether a specific version has been negotiated yet or not. Although it is only ever called from one place where we are always in `*_ANY_VERSION`.

Reply:
The relocated code explains that the version may already be set during renegotiation.